library/Ivoz/Core/Infrastructure/Domain/Service/Cgrates/AbstractBalanceService.php

```diff
@@ -62,12 +64,21 @@ protected function debitBalance(EntityInterface $entity, float $amount)
     }
 
     /**
-     * @param string $tenant
-     * @param array $accountIds
+     * @param int $brandId
+     * @param array $ids
+     * @param string $accountPrefix
      * @return array|mixed
      */
-    protected function getAccountsBalances(string $tenant, array $accountIds)
+    protected function getAccountsBalances(int $brandId, array $ids, string $accountPrefix)
     {
+        $tenant = self::TENANT_PREFIX . $brandId;
+        $accountIds = array_map(
+            function ($companyId) use ($accountPrefix) {
+                return $accountPrefix . $companyId;
+            },
+            $ids
+        );
+
```

Comment:
If you pass _BrandInterface_ instead of brandId here you can use $brand->getCgrTenant() method so you no longer require tenant prefix constant.

Also, $ids array would be better if there were $accountIds (with the prefix included) instead of adding an extra parameter. AccountIds can be obtained by calling getCgrSubject available from both Companies and Carriers.

I assume this is not implemented that way because this is called from workers that only have ids to work with and you prefer to avoid a repository call to fetch the whole entity but leaving all the prefix stuff encapsulated into Brand->getCgrTenant and Company|Carrier->getCgrSubject would be ideal for maintenance.